eclipse-theia/theia
# fix(task): Align task group validation with VS Code

#### What it does

In VS Code, the following is considered correct in a `tasks.json`:
```
"group": {"kind": "build","isDefault": false}
```

In Theia, however, this is considered an error as the enumeration in the JSON Schema restricts valid values to `isDefault: true`. This also prevents such tasks to be listed correctly when e.g. executing the command "Run Build Task".

Contributed on behalf of STMicroelectronics.

Fixes https://github.com/eclipse-theia/theia/issues/12917

#### How to test

* Create a `tasks.json` with the following content:
```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "echo",
            "type": "shell",
            "command": "echo Hello",
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```
* Observe how it is considered valid and automatically executes when the user executes "Run Build Task"
* Change the task to below and observe how it is still considered valid and shows up as a suggestion when executing "Run Build Task".
```json
        {
            "label": "echo",
            "type": "shell",
            "command": "echo Hello",
            "group": {
                "kind": "build",
                "isDefault": false
            }
        }
```

#### Follow-ups

There is another difference to the behavior of VS Code:
If there is only one task in the group, it'll be considered the default task and executed automatically. In Theia, the user is still prompted which task to execute even though there is just one. Not sure if we want to change that though.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
